dll/win32/browseui/shellfind/CSearchBar.cpp

```diff
@@ -131,6 +137,34 @@ LRESULT CSearchBar::OnSearchButtonClicked(WORD wNotifyCode, WORD wID, HWND hWndC
         return 0;
     }
 
+    // See if we have a szFileName, our first character is not an asterisk,
+    // and we have room in MAX_PATH to add an asterisk at the beginning of szFileName then add it
+    if(wcscmp(pSearchStart->szFileName, L"") != 0)
+        {
+        len = wcslen(pSearchStart->szFileName);
+        if(len > 0)
+            {
+            ptrchar = pSearchStart->szFileName[len - 1];
+            memcpy(endchar, &ptrchar, 2);
+            ptrchar = pSearchStart->szFileName[0];
+            memcpy(startchar, &ptrchar, 2);
+            if ((len < MAX_PATH) && (wcscmp(startchar, L"*") != 0))
+                {
+                    memmove(&pSearchStart->szFileName[1], &pSearchStart->szFileName[0],
+                            len * sizeof(WCHAR) + sizeof(WCHAR));
```

Comment:
I have the impression you are going to read outside the buffer there...

Reply:
The source is ...[0] and the destination is ...[1], so we are moving the characters to the right one character position which will leave a position at the start of the string that we can put our asterisk. The move size is calculated based on the length and we check that this is less than "MAX_PATH", so this should mean that we have at least one character position at the end. For me this seems to confirm that we will not write outside the buffer.